contributors/design-proposals/controller-ref.md

```diff
@@ -1,100 +1,304 @@
+controllers.
+
+## The Three Laws of Controllers
+
+All controllers that manage collections of objects should obey the following
+rules.
+
+1. **Take ownership**
+
+   A controller must claim *ownership* of any objects it creates by adding a
+   ControllerRef, and may also claim ownership of an object it didn't create,
+   as long as the object has no existing ControllerRef (i.e. it is an *orphan*).
+
+1. **Don't interfere**
+
+   A controller may not take any action (e.g. edit/scale/delete) on an object it
+   does not own, except to [*adopt*](#adoption) the object if allowed by the
+   First Law.
+
+1. **Don't share**
+
+   A controller should not count an object it does not own toward satisfying its
+   desired state (e.g. a certain number of replicas), although it may include
+   the object in plans to achieve its desired state (e.g. through adoption)
+   as long as such plans do not conflict with the First or Second Laws.
+
+## Adoption
+
+If a controller finds an orphaned object that matches its selector, it should
+try to adopt the object by adding a ControllerRef.
+
+Multiple controllers may race to adopt a given object, but only one can win
+by being the first to add a ControllerRef to the object's OwnerReferences list.
+The losers will see their adoptions fail due to a validation error as explained
+[above](#api).
+
+If a controller has a non-nil `DeletionTimestamp`, it must not attempt adoption
+or take any other actions except updating its `Status`.
+This prevents readoption of objects orphaned by the [orphan finalizer](garbage-collection.md#part-ii-the-orphan-finalizer)
+during deletion of the controller.
+
+## Orphaning
+
+When a controller is deleted, the objects it owns will either be orphaned or
+deleted according to the normal [Garbage Collection](garbage-collection.md)
+behavior, based on OwnerReferences.
+
+In addition, if a controller finds that it owns an object that no longer matches
+its selector, it should orphan the object by removing itself from the object's
+OwnerReferences list. Since ControllerRef is just a special type of
+OwnerReference, this also means the ControllerRef is removed.
+
+## Watches
+
+Many controller managers use watches to *sync* each controller (prompting it to
+reconcile desired and actual state) as soon as a relevant event occurs for one
+of its controlled objects, as well as to let controllers wait for asynchronous
+operations to complete on those objects.
+The controller manager subscribes to a stream of events about controlled objects
+and routes each event to a particular controller.
+
+Previously, the controller manager used only label selectors to decide which
+controller to route an event to. If multiple controllers had overlapping
+selectors, events might be misrouted, causing the wrong controllers to sync,
+or causing some controllers to freeze because they keep waiting for an event
+that already happened.
+
+Some controller managers introduced a workaround that used the controller
```

Comment:
I associate "controller manager" with the daemon itself, as per the definition in https://kubernetes.io/docs/admin/kube-controller-manager/. I think of them as controllers and sync/control loops per instance. I'm not sure if "controller manager" is terminology that is used in other proposals.